darts/models/forecasting/forecasting_model.py

```diff
@@ -1032,6 +1047,9 @@ def retrain_func(
                     verbose=verbose,
                     predict_likelihood_parameters=predict_likelihood_parameters,
                 )
+                if is_scalar_used:
```

Comment:
Comment for line 1040 (just before model._predict_wrapper()): We could allow support for passing a transformer even if retrain=False. We would just assume that the transformer was fit before, or that it doesn't require fitting.

                # also apply for covariates, here we can assume that the scaler must have been fit before or not

```
                # require fitting at all
                if scaler is not None and not train_transfomed:
                    train_series = scaler.transform(train_series)
```

Reply:
Makes sense to add allow that as a feature! Added it to the code, line 1196 onwards